bessctl/conf/port/vhost/launch_container.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import os
+import sys
+import subprocess
+import time
+import shlex
+
+# How many cores we reserve for vSwitches?
+# If set to 2, Containers will run on core 2, 3, 4, ..., skipping core 0-1.
+VM_START_CPU = int(os.getenv('VM_START_CPU', '1'))
+VM_MEM_SOCKET = int(os.getenv('VM_MEM_SOCKET', '0'))
+
+HUGEPAGES_PATH = os.getenv('HUGEPAGES_PATH', '/dev/hugepages')
+
+# "io retry" gives better throughput as the VM will not touch the packet
+# payload, but it is somewhat unrealistic...
+FWD_MODE = os.getenv('FWD_MODE', 'macswap retry')
+
+# per container configuration
+# NUM_CPUS cores are used for forwarding
+NUM_VCPUS = int(os.getenv('VM_VCPUS', '2'))
+NUM_VPORTS = int(os.getenv('BESS_PORTS', '2'))
+NUM_QUEUES = int(os.getenv('BESS_QUEUES', '1'))
+
+VERBOSE = int(os.getenv('VERBOSE', '0'))
+
+SOCKDIR = '/tmp/bessd'
+IMAGE = 'nefelinetworks/bess_build'
+
+
+def launch(cid):
+    print('Running container {} as a forwarder'.format(cid))
+    first_cpu = VM_START_CPU + cid * NUM_VCPUS
+    last_cpu = first_cpu + NUM_VCPUS - 1
+    eal_opts = '--file-prefix=vnf{} --no-pci -m 256 -l 0,{}-{}'.format(
+        cid, first_cpu, last_cpu)
+
+    for port_id in range(NUM_VPORTS):
+        sockpath = '{}/vhost_user{}_{}.sock'.format(SOCKDIR, cid, port_id)
+        eal_opts += ' --vdev=virtio_user{},path={}'.format(port_id, sockpath)
+
+    testpmd_opts = '-i --burst=64 --txd=1024 --rxd=1024 --disable-hw-vlan ' \
```

Comment:
trivial: might be nice to share some of this code with the launch_vm.py code (though it looks like only the testpmd setup itself is the same).